packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -389,41 +403,33 @@ export class Key extends KeyBase {
     }
   }
 
-  private allowAccountIdentitiesToControl() {
+  /**
+   * Adds the default key policy to the key. This policy gives the AWS account (root user) full access to the CMK,
+   * which reduces the risk of the CMK becoming unmanageable and enables IAM policies to allow access to the CMK.
+   * This is the same policy that is default when creating a Key via the KMS API or Console.
+   * @see https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
+   */
+  private grantDefaultKeyPolicy() {
     this.addToResourcePolicy(new iam.PolicyStatement({
       resources: ['*'],
       actions: ['kms:*'],
       principals: [new iam.AccountRootPrincipal()],
     }));
-
   }
+
   /**
-   * Let users or IAM policies from this account admin this key.
+   * Grants the account admin privileges -- not full account access -- plus the GenerateDataKey action.
+   * The GenerateDataKey action was added for interop with S3 in https://github.com/aws/aws-cdk/issues/3458.
+   *
+   * This policy is discouraged and deprecated by the '@aws-cdk/aws-kms:defaultKeyPolicies' feature flag.
+   *
    * @link https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
-   * @link https://aws.amazon.com/premiumsupport/knowledge-center/update-key-policy-future/
+   * @deprecated
    */
-  private allowAccountToAdmin() {
-    const actions = [
-      'kms:Create*',
-      'kms:Describe*',
-      'kms:Enable*',
-      'kms:List*',
-      'kms:Put*',
-      'kms:Update*',
-      'kms:Revoke*',
-      'kms:Disable*',
-      'kms:Get*',
-      'kms:Delete*',
-      'kms:ScheduleKeyDeletion',
-      'kms:CancelKeyDeletion',
-      'kms:GenerateDataKey',
-      'kms:TagResource',
-      'kms:UntagResource',
-    ];
-
+  private grantAccountAdminPrivilegesPlusGenerateDataKey() {
```

Comment:
Same note here - I would avoid using the word `grant`, for the same reasons.

Also, the principal here is the same as in `grantDefaultKeyPolicy()`, but we don't call it the same in the two methods.

`addAdminAndGenDataKeyForRootPrincipalToResourcePolicy()`?

Reply:
I hate all these long names that basically describe what the method does with `And`s. Isn't it more useful to give it a name that symbolically represents it and treats the method as a single chunk? You can dive into what the method does if you need to (it's just a Command-click away), but chances are you don't.

I would vote for something simple like:

```ts
this.addLegacyAdminPolicy();
```

If you want to represent that it was broken in the first place---

---well first ask yourself, is that important enough to surface in the name? You can always add in an implementation comment:

```
// GenerateDataKey should never have been here in the first place, but we were young and didn't know any better
```

And that perfectly suffices. If you REALLY think you need to deter people from calling this in the future:

```
this.addIncorrectLegacyAdminPolicy();
```